### PR TITLE
Feat: Added button component.

### PR DESCRIPTION
**Pull request by coldpigli**
Includes Primary, Outlined, Link and Floating Action Buttons.

contained-button
outlined-button
Link Button
fab-button

### CODE REVIEW

File: css/components/button.css [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+.btn-failure-outline{
+    background-color: transparent;
+    color: var(--cherry);
+    border-color: var(--cherry);
+}
+
+.btn-failure-outline:hover{
+    background-color: var(--cherry-light-opaque);
+    color: var(--cherry);
```
Comment (vivekmehto) [inline css/components/button.css]: avoid repeated code.

File: css/components/button.css [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+.btn-secondary-outline{
+    background-color: transparent;
+    color: var(--secondary);
+    border-color: var(--secondary);
+}
+
+.btn-secondary-outline:hover{
+    background-color: var(--secondary-opaque);
+    color: var(--secondary);
```
Comment (vivekmehto) [inline css/components/button.css]: color and border color don't change in hover state.

File: css/components/button.css [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+.btn:hover{
+    border-color: var(--gray-hover);
+    background-color: var(--gray-hover);
+}
+
+.btn-primary{
+    border-color: var(--primary-light);
+    background-color: var(--primary-light);
+    color: var(--background-day);
```
Comment (vivekmehto) [inline css/components/button.css]: You can give color to the btn class. It is repeated in 2-3 class.

File: css/components/button.css [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+.btn-primary-outline{
+    background-color: transparent;
+    color: var(--primary);
+    border-color: var(--primary);
+}
+
+.btn-primary-outline:hover{
+    background-color: var(--primary-opaque);
+    color: var(--primary);
```
Comment (vivekmehto) [inline css/components/button.css:73]: This one also.

**Review by vivekmehto — COMMENTED** [review]
Avoid repeating the same code. Other than that everything looks good.

File: css/components/button.css (hunk 3 above)
Comment (coldpigli) [reply css/components/button.css]: > You can give color to the btn class. It is repeated in 2-3 class.

Hi thanks for the heads up. It  had totally gone unnoticed :D thanks. I have refactored the code.

File: css/components/button.css (hunk 2 above)
Comment (coldpigli) [reply css/components/button.css]: Removed redundant code. Thanks

**Review by krishanu2911 — COMMENTED** [review]
Good that you have used meaningful class names also used CSS variables
Everything seems good to me

**Pull request merged**